### PR TITLE
Fix (and test) some /server/audit query parameters

### PR DESCRIPTION
**Pull request by dbutenhof**
PBENCH-1118

On a whim I created an admin user (which we don't routinely do) and tried some audit log queries. Turns out that I wasn't really thinking clearly when I chose "obvious" query parameter names and used them directly in the call to `Audit.query`, because the parameter names don't always match the defined columns or `kwargs`.

This fixes the bad names, and adds a bunch of unit test cases to confirm them, which includes some refactoring to do more thorough result checks without a mass of repeated text lines.

### CODE REVIEW

File: lib/pbench/test/unit/server/test_server_audit.py [hunk 1]
```diff
@@ -1,19 +1,30 @@
...
 
 
 class TestServerAudit:
+    audits = []
+
     @pytest.fixture()
-    def query_get(self, client, server_config, pbench_admin_token):
+    def query_get(
+        self,
+        client: FlaskClient,
```
Comment (dbutenhof) [inline lib/pbench/test/unit/server/test_server_audit.py:24]: This is actually a bit intriguing. I added a reference to `AuditReason` and normally `vscode` is quite obliging on a `ctrl-.` to add imports, but didn't see it. It *did* have an option I'd never noticed before to add types for all fixtures (yes, specifically "fixtures"), and I accidentally let go of the button hovering over it. So it went ahead. And I had to search for how to import a few (like `FlaskClient`, which isn't trivially obvious), but I decided to leave it...

File: lib/pbench/test/unit/server/test_server_audit.py [hunk 2]
```diff
@@ -58,52 +76,86 @@ def make_audits(self, client, create_user) -> str:
...
         assert audits[1]["status"] == "SUCCESS"
         assert audits[1]["operation"] == "CREATE"
         assert audits[1]["name"] == "first"
-        assert audits[1]["user_id"] == expected_user_id
+        assert audits[1]["user_id"] == str(create_user.id)
         assert audits[1]["user_name"] == "test"
+        assert audits[1]["object_id"] is None
+        assert audits[1]["object_name"] is None
+        assert audits[1]["object_type"] is None
+        assert audits[1]["reason"] is None
```
Comment (webbnh) [inline lib/pbench/test/unit/server/test_server_audit.py:107]: Should we be checking `"user_name"` here?  (Ditto for entries 2 and 3?)

File: lib/pbench/test/unit/server/test_server_audit.py [hunk 3]
```diff
@@ -58,52 +76,86 @@ def make_audits(self, client, create_user) -> str:
...
+        TestServerAudit.audits = audits
+        return user_id
+
+    def check_audits(self, actual: list[Audit], expected: list[int]):
+        assert len(actual) == len(expected)
+        ai = 0
+
+        for e in expected:
+            assert actual[ai] == self.audits[e]
+            ai += 1
```
Comment (webbnh) [inline lib/pbench/test/unit/server/test_server_audit.py:138]: Alternately,
```python
        for ai, e in enumerate(expected):
            assert actual[ai] == self.audits[e]
```

**Review by webbnh — APPROVED** [review]
Looks OK to me; just a nit, question, and suggestion.

File: lib/pbench/server/api/resources/server_audit.py [hunk 4]
```diff
@@ -41,13 +41,14 @@ def __init__(self, config: PbenchServerConfig):
...
+                    Parameter("object_name", ParamType.STRING),
+                    Parameter("object_type", ParamType.KEYWORD, enum=AuditType),
                     Parameter("operation", ParamType.KEYWORD, enum=OperationCode),
                     Parameter("reason", ParamType.KEYWORD, enum=AuditReason),
                     Parameter("start", ParamType.DATE),
                     Parameter("status", ParamType.KEYWORD, enum=AuditStatus),
-                    Parameter("type", ParamType.KEYWORD, enum=AuditType),
-                    Parameter("user", ParamType.USER),
                     Parameter("user_id", ParamType.STRING),
+                    Parameter("user_name", ParamType.STRING),
```
Comment (npalaska) [inline lib/pbench/server/api/resources/server_audit.py:51]: Is there a need to keep both?

**Review by npalaska — APPROVED** [review]

File: lib/pbench/test/unit/server/test_server_audit.py (hunk 3 above)
Comment (dbutenhof) [reply lib/pbench/test/unit/server/test_server_audit.py:138]: I knew there was a way to do that ... I couldn't remember `enumerate` at the time.

File: lib/pbench/test/unit/server/test_server_audit.py [hunk 5]
```diff
@@ -58,52 +76,86 @@ def make_audits(self, client, create_user) -> str:
...
+                root=root, status=AuditStatus.FAILURE, reason=AuditReason.PERMISSION
+            )
         response = query_get(expected_status=HTTPStatus.OK)
         audits = response.json
-        assert len(audits) == 4
         assert audits[0]["status"] == "BEGIN"
         assert audits[0]["operation"] == "CREATE"
         assert audits[0]["name"] == "first"
-        assert audits[0]["user_id"] == expected_user_id
+        assert audits[0]["user_id"] == str(create_user.id)
```
[the remark this answers is not in the document]
Comment (webbnh) [reply lib/pbench/test/unit/server/test_server_audit.py:92]: *sigh*

File: lib/pbench/test/unit/server/test_server_audit.py (hunk 2 above)
Comment (webbnh) [reply lib/pbench/test/unit/server/test_server_audit.py:107]: I think perhaps it is the addition of line 97 which is the problem.  (Does it add anything over line 93?)

**Review by npalaska — APPROVED** [review]

**Pull request merged**